{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "2d3d61c1-42c6-48aa-a19c-d55ae9c62679",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/dml/JupyterLab/hmmer-3.4/tutorial\n",
      "HMMBUILD PATH: /Users/dml/.local/share/mamba/envs/myjlabenv/bin/hmmbuild\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "# Get the path to the active environment's 'bin' directory\n",
    "bin_path = os.path.join(sys.prefix, 'bin')\n",
    "hmmbuild_path = os.path.join(bin_path, 'hmmbuild')\n",
    "\n",
    "print(os.getcwd())\n",
    "print(f\"HMMBUILD PATH: {hmmbuild_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "4c47d4fa-2d10-4f9b-a2d8-5752483593fc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "Usage: hmmbuild [-options] <hmmfile_out> <msafile>\n",
      "\n",
      "Basic options:\n",
      "  -h     : show brief help on version and usage\n",
      "  -n <s> : name the HMM <s>\n",
      "  -o <f> : direct summary output to file <f>, not stdout\n",
      "  -O <f> : resave annotated, possibly modified MSA to file <f>\n",
      "\n",
      "Options for selecting alphabet rather than guessing it:\n",
      "  --amino : input alignment is protein sequence data\n",
      "  --dna   : input alignment is DNA sequence data\n",
      "  --rna   : input alignment is RNA sequence data\n",
      "\n",
      "Alternative model construction strategies:\n",
      "  --fast           : assign cols w/ >= symfrac residues as consensus  [default]\n",
      "  --hand           : manual construction (requires reference annotation)\n",
      "  --symfrac <x>    : sets sym fraction controlling --fast construction  [0.5]\n",
      "  --fragthresh <x> : if L <= x*alen, tag sequence as a fragment  [0.5]\n",
      "\n",
      "Alternative relative sequence weighting strategies:\n",
      "  --wpb     : Henikoff position-based weights  [default]\n",
      "  --wgsc    : Gerstein/Sonnhammer/Chothia tree weights\n",
      "  --wblosum : Henikoff simple filter weights\n",
      "  --wnone   : don't do any relative weighting; set all to 1\n",
      "  --wgiven  : use weights as given in MSA file\n",
      "  --wid <x> : for --wblosum: set identity cutoff  [0.62]  (0<=x<=1)\n",
      "\n",
      "Alternative effective sequence weighting strategies:\n",
      "  --eent       : adjust eff seq # to achieve relative entropy target  [default]\n",
      "  --eclust     : eff seq # is # of single linkage clusters\n",
      "  --enone      : no effective seq # weighting: just use nseq\n",
      "  --eset <x>   : set eff seq # for all models to <x>\n",
      "  --ere <x>    : for --eent: set minimum rel entropy/position to <x>\n",
      "  --esigma <x> : for --eent: set sigma param to <x>  [45.0]\n",
      "  --eid <x>    : for --eclust: set fractional identity cutoff to <x>  [0.62]\n",
      "\n",
      "Alternative prior strategies:\n",
      "  --pnone    : don't use any prior; parameters are frequencies\n",
      "  --plaplace : use a Laplace +1 prior\n",
      "\n",
      "Handling single sequence inputs:\n",
      "  --singlemx    : use substitution score matrix for single-sequence inputs\n",
      "  --mx <s>      : substitution score matrix (built-in matrices, with --singlemx)\n",
      "  --mxfile <f>  : read substitution score matrix from file <f> (with --singlemx)\n",
      "  --popen <x>   : force gap open prob. (w/ --singlemx, aa default 0.02, nt 0.031)\n",
      "  --pextend <x> : force gap extend prob. (w/ --singlemx, aa default 0.4, nt 0.75)\n",
      "\n",
      "Control of E-value calibration:\n",
      "  --EmL <n> : length of sequences for MSV Gumbel mu fit  [200]  (n>0)\n",
      "  --EmN <n> : number of sequences for MSV Gumbel mu fit  [200]  (n>0)\n",
      "  --EvL <n> : length of sequences for Viterbi Gumbel mu fit  [200]  (n>0)\n",
      "  --EvN <n> : number of sequences for Viterbi Gumbel mu fit  [200]  (n>0)\n",
      "  --EfL <n> : length of sequences for Forward exp tail tau fit  [100]  (n>0)\n",
      "  --EfN <n> : number of sequences for Forward exp tail tau fit  [200]  (n>0)\n",
      "  --Eft <x> : tail mass for Forward exponential tail tau fit  [0.04]  (0<x<1)\n",
      "\n",
      "Other options:\n",
      "  --cpu <n>          : number of parallel CPU workers for multithreads  [2]\n",
      "  --mpi              : run as an MPI parallel program\n",
      "  --stall            : arrest after start: for attaching debugger to process\n",
      "  --informat <s>     : assert input alifile is in format <s> (no autodetect)\n",
      "  --seed <n>         : set RNG seed to <n> (if 0: one-time arbitrary seed)  [42]\n",
      "  --w_beta <x>       : tail mass at which window length is determined\n",
      "  --w_length <n>     : window length \n",
      "  --maxinsertlen <n> : pretend all inserts are length <= <n>\n"
     ]
    }
   ],
   "source": [
    "!/Users/dml/.local/share/mamba/envs/myjlabenv/bin/hmmbuild -h # provides brief help documentation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "6980cd3e-3a8b-49d9-9a91-c4d85b6f00e8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             globins4.sto\n",
      "# output HMM file:                  globins4.hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     globins4                 4   171   149     0.96  0.589 \n",
      "\n",
      "# CPU time: 0.04u 0.00s 00:00:00.04 Elapsed: 00:00:00.04\n"
     ]
    }
   ],
   "source": [
    "!/Users/dml/.local/share/mamba/envs/myjlabenv/bin/hmmbuild globins4.hmm globins4.sto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "8fa0594e-7e10-4492-9924-8ce9d3951f96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# hmmsearch :: search profile(s) against a sequence database\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# query HMM file:                  globins4.hmm\n",
      "# target sequence database:        globins45.fa\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "Query:       globins4  [M=149]\n",
      "Scores for complete sequences (score includes all domains):\n",
      "   --- full sequence ---   --- best 1 domain ---    -#dom-\n",
      "    E-value  score  bias    E-value  score  bias    exp  N  Sequence   Description\n",
      "    ------- ------ -----    ------- ------ -----   ---- --  --------   -----------\n",
      "    1.9e-67  217.7   0.1    2.1e-67  217.5   0.1    1.0  1  HBB_MANSP   \n",
      "    1.3e-66  215.0   0.2    1.4e-66  214.9   0.2    1.0  1  HBB_CALAR   \n",
      "    2.6e-66  214.0   3.0    2.9e-66  213.9   3.0    1.0  1  MYG_ESCGI   \n",
      "    7.3e-65  209.3   0.2    8.1e-65  209.2   0.2    1.0  1  HBB_URSMA   \n",
      "    2.3e-64  207.7   0.6    2.5e-64  207.6   0.6    1.0  1  HBB_RABIT   \n",
      "      8e-64  205.9   1.4    8.9e-64  205.8   1.4    1.0  1  HBA_PONPY   \n",
      "    1.3e-63  205.2   1.2    1.5e-63  205.1   1.2    1.0  1  MYG_HORSE   \n",
      "    7.7e-63  202.8   1.2    8.5e-63  202.6   1.2    1.0  1  HBB_SPECI   \n",
      "    4.5e-62  200.3   0.6      5e-62  200.1   0.6    1.0  1  HBA_MACFA   \n",
      "    5.2e-62  200.1   0.7    5.8e-62  199.9   0.7    1.0  1  HBB_SPETO   \n",
      "    6.6e-62  199.7   0.1    7.4e-62  199.6   0.1    1.0  1  HBB_SUNMU   \n",
      "    1.1e-61  199.0   0.5    1.3e-61  198.8   0.5    1.0  1  HBA_MACSI   \n",
      "    1.2e-61  198.9   0.1    1.3e-61  198.8   0.1    1.0  1  HBB_TRIIN   \n",
      "    9.3e-61  196.0   0.1      1e-60  195.9   0.1    1.0  1  HBB_EQUHE   \n",
      "    1.5e-60  195.3   1.0    1.7e-60  195.2   1.0    1.0  1  HBA2_GALCR  \n",
      "    2.1e-60  194.9   1.7    2.4e-60  194.7   1.7    1.0  1  MYG_LYCPI   \n",
      "    2.2e-60  194.8   0.3    2.4e-60  194.7   0.3    1.0  1  MYG_PROGU   \n",
      "    3.9e-60  194.0   1.1    4.3e-60  193.9   1.1    1.0  1  MYG_SAISC   \n",
      "    8.8e-60  192.8   0.4    9.7e-60  192.7   0.4    1.0  1  HBB_TACAC   \n",
      "    3.5e-59  190.9   0.6    3.8e-59  190.8   0.6    1.0  1  HBE_PONPY   \n",
      "    5.7e-59  190.2   0.4    6.3e-59  190.1   0.4    1.0  1  HBA_MESAU   \n",
      "    8.7e-59  189.6   0.8    9.7e-59  189.5   0.8    1.0  1  HBA2_BOSMU  \n",
      "    2.3e-58  188.3   0.0    2.5e-58  188.1   0.0    1.0  1  HBB_TUPGL   \n",
      "    4.9e-58  187.2   0.2    5.4e-58  187.0   0.2    1.0  1  HBA_AILME   \n",
      "    5.7e-58  187.0   0.3    6.3e-58  186.8   0.3    1.0  1  MYG_MOUSE   \n",
      "    8.2e-58  186.5   0.0    9.1e-58  186.3   0.0    1.0  1  HBB_ORNAN   \n",
      "    1.3e-57  185.8   0.1    1.5e-57  185.6   0.1    1.0  1  HBB_COLLI   \n",
      "    2.1e-56  181.9   0.1    2.3e-56  181.7   0.1    1.0  1  HBB_LARRI   \n",
      "      4e-56  181.0   0.7    4.4e-56  180.8   0.7    1.0  1  HBA_PAGLA   \n",
      "    1.2e-55  179.4   0.2    1.4e-55  179.3   0.2    1.0  1  HBA_ERIEU   \n",
      "    1.2e-55  179.4   0.1    1.4e-55  179.2   0.1    1.0  1  HBA_PROLO   \n",
      "    2.3e-52  168.8   0.1    2.5e-52  168.7   0.1    1.0  1  HBB1_VAREX  \n",
      "    3.8e-52  168.1   0.2    4.2e-52  167.9   0.2    1.0  1  HBAZ_HORSE  \n",
      "    2.2e-51  165.6   0.5    2.4e-51  165.5   0.5    1.0  1  HBA_FRAPO   \n",
      "    3.3e-51  165.0   0.5    3.6e-51  164.9   0.5    1.0  1  HBA_PHACO   \n",
      "    6.3e-50  160.9   0.6      7e-50  160.7   0.6    1.0  1  HBA_ANSSE   \n",
      "    6.9e-50  160.7   0.1    7.6e-50  160.6   0.1    1.0  1  HBAD_PASMO  \n",
      "    9.5e-50  160.3   0.4    1.1e-49  160.1   0.4    1.0  1  HBA_TRIOC   \n",
      "    1.4e-49  159.8   0.5    1.5e-49  159.6   0.5    1.0  1  HBAD_CHLME  \n",
      "    1.4e-49  159.7   0.3    1.6e-49  159.6   0.3    1.0  1  HBA_COLLI   \n",
      "    2.9e-49  158.7   0.3    3.1e-49  158.6   0.3    1.0  1  HBBL_RANCA  \n",
      "    2.1e-48  155.9   1.2    2.3e-48  155.8   1.2    1.0  1  HBB2_XENTR  \n",
      "    6.8e-44  141.3   0.1    7.6e-44  141.1   0.1    1.0  1  HBA4_SALIR  \n",
      "    2.1e-37  120.2   0.4    2.3e-37  120.1   0.4    1.0  1  MYG_MUSAN   \n",
      "      1e-36  118.0   0.0    1.1e-36  117.9   0.0    1.0  1  HBB2_TRICR  \n",
      "\n",
      "\n",
      "Domain annotation for each sequence (and alignments):\n",
      ">> HBB_MANSP  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  217.5   0.1   2.1e-67   2.1e-67       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 217.5 bits;  conditional E-value: 2.1e-67\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+L+++ekt+v+++W+kv  +v+evG+++L rl+v+yP+tq+fF++F+dLs +d+++g+++vk+Hgkkvl+A+sd+l++ld +l++++++LselH++kl+\n",
      "  HBB_MANSP   1 VHLTPEEKTAVTTLWGKV--NVDEVGGEALGRLLVVYPWTQRFFDSFGDLSSPDAVMGNPKVKAHGKKVLGAFSDGLNHLD-NLKGTFAQLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++fkll++vlv+vla++++keft++vqaa++K++a+va+ala+kY+\n",
      "  HBB_MANSP  98 VDPENFKLLGNVLVCVLAHHFGKEFTPQVQAAYQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBB_CALAR  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  214.9   0.2   1.4e-66   1.4e-66       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 214.9 bits;  conditional E-value: 1.4e-66\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+L+ +ek++v+a+W+kv  +v+evG+++L rl+v+yP+tq+fFe+F+dLst+d+++++++vk+Hgkkvl+A+sd+l +ld +l++++++LselH++kl+\n",
      "  HBB_CALAR   1 VHLTGEEKSAVTALWGKV--NVDEVGGEALGRLLVVYPWTQRFFESFGDLSTPDAVMNNPKVKAHGKKVLGAFSDGLTHLD-NLKGTFAHLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ll++vlv+vla++++keft+ vqaa++K++a+va+ala+kY+\n",
      "  HBB_CALAR  98 VDPENFRLLGNVLVCVLAHHFGKEFTPVVQAAYQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> MYG_ESCGI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  213.9   3.0   2.9e-66   2.9e-66       2     149 .]       1     147 [.       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 213.9 bits;  conditional E-value: 2.9e-66\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+ae++ v+++Wakveadv+++G+diL+rlf+ +Pet+e+F+kFk+L+te+e+k+s+d+kkHg++vl+Al+ +l+k ++++eaelk+L+++Ha+k+k+\n",
      "  MYG_ESCGI   1 VLSDAEWQLVLNIWAKVEADVAGHGQDILIRLFKGHPETLEKFDKFKHLKTEAEMKASEDLKKHGNTVLTALGGILKK-KGHHEAELKPLAQSHATKHKI 99 \n",
      "                69****************************************************************************.99******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                ++ky++++s+++++vl++r+p++f+ad+qaa++K+l+l++k++a+kYk\n",
      "  MYG_ESCGI 100 PIKYLEFISDAIIHVLHSRHPGDFGADAQAAMNKALELFRKDIAAKYK 147\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBB_URSMA  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  209.2   0.2   8.1e-65   8.1e-65       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 209.2 bits;  conditional E-value: 8.1e-65\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+L+ +ek+ v+ +W+kv  +v+evG+++L rl+v+yP+tq+fF++F+dLs +d+++++++vk+Hgkkvl+ +sd+l++ld +l++++++LselH++kl+\n",
      "  HBB_URSMA   1 VHLTGEEKSLVTGLWGKV--NVDEVGGEALGRLLVVYPWTQRFFDSFGDLSSADAIMNNPKVKAHGKKVLNSFSDGLKNLD-NLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++fkll++vlv+vla++++keft++vqaa++K++a+va+ala+kY+\n",
      "  HBB_URSMA  98 VDPENFKLLGNVLVCVLAHHFGKEFTPQVQAAYQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBB_RABIT  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  207.6   0.6   2.5e-64   2.5e-64       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 207.6 bits;  conditional E-value: 2.5e-64\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls +ek++v+a+W+kv  +veevG+++L rl+v+yP+tq+fFe+F+dLs +++++++++vk+Hgkkvl A+s++l +ld +l++++++LselH++kl+\n",
      "  HBB_RABIT   1 VHLSSEEKSAVTALWGKV--NVEEVGGEALGRLLVVYPWTQRFFESFGDLSSANAVMNNPKVKAHGKKVLAAFSEGLSHLD-NLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ll++vlv vl ++++keft++vqaa++K++a+va+ala+kY+\n",
      "  HBB_RABIT  98 VDPENFRLLGNVLVIVLSHHFGKEFTPQVQAAYQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBA_PONPY  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  205.8   1.4   8.9e-64   8.9e-64       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 205.8 bits;  conditional E-value: 8.9e-64\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+kt+vk++W+kv+a+++++Ga++L+r+f+s+P+t+++F++F dLs     +gsa+vk Hgkkv+dAl++a+a++d +++++l+aLs+lHa+kl+v\n",
      "  HBA_PONPY   1 VLSPADKTNVKTAWGKVGAHAGDYGAEALERMFLSFPTTKTYFPHF-DLS-----HGSAQVKDHGKKVADALTNAVAHVD-DMPNALSALSDLHAHKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+laa+lp+eft++v+a+l+K+la+v+++l+skY+\n",
      "  HBA_PONPY  94 DPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> MYG_HORSE  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  205.1   1.2   1.5e-63   1.5e-63       3     149 .]       2     147 ..       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 205.1 bits;  conditional E-value: 1.5e-63\n",
      "   globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvd 102\n",
      "                Ls++e+++v++vW+kvead++++G+++L+rlf+ +Pet+e+F+kFk+L+te+e+k+s+d+kkHg  vl+Al+ +l+k ++++eaelk+L+++Ha+k+k++\n",
      "  MYG_HORSE   2 LSDGEWQQVLNVWGKVEADIAGHGQEVLIRLFTGHPETLEKFDKFKHLKTEAEMKASEDLKKHGTVVLTALGGILKK-KGHHEAELKPLAQSHATKHKIP 100\n",
      "                89***************************************************************************.99******************** PP\n",
      "\n",
      "   globins4 103 pkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                +ky++++s+++++vl++++p++f+ad+q+a+ K+l+l+++++a+kYk\n",
      "  MYG_HORSE 101 IKYLEFISDAIIHVLHSKHPGNFGADAQGAMTKALELFRNDIAAKYK 147\n",
      "                **********************************************7 PP\n",
      "\n",
      ">> HBB_SPECI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  202.6   1.2   8.5e-63   8.5e-63       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 202.6 bits;  conditional E-value: 8.5e-63\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls++ek+++ ++W+kv  +++evGa++L rl+v+yP+tq+fF++F+dLs + +++g+a+vk+Hgkkv+d +s++l++ld +l++++++LselH++kl+\n",
      "  HBB_SPECI   1 VHLSDGEKNAISTAWGKV--HAAEVGAEALGRLLVVYPWTQRFFDSFGDLSSASAVMGNAKVKAHGKKVIDSFSNGLKHLD-NLKGTFASLSELHCDKLH 97 \n",
      "                69****************..99***********************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++fkll++++v v+a++l+k+ft+++qaa++K++a+va+ala+kY+\n",
      "  HBB_SPECI  98 VDPENFKLLGNMIVIVMAHHLGKDFTPEAQAAFQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBA_MACFA  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  200.1   0.6     5e-62     5e-62       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 200.1 bits;  conditional E-value: 5e-62\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+kt+vka+W+kv+ +++e+Ga++L+r+f+s+P+t+++F++F dLs     +gsa+vk Hgkkv+dAl+ a+ ++d +++++l+aLs+lHa+kl+v\n",
      "  HBA_MACFA   1 VLSPADKTNVKAAWGKVGGHAGEYGAEALERMFLSFPTTKTYFPHF-DLS-----HGSAQVKGHGKKVADALTLAVGHVD-DMPQALSALSDLHAHKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+laa+lp+eft++v+a+l+K+la+v+++l+skY+\n",
      "  HBA_MACFA  94 DPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBB_SPETO  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  199.9   0.7   5.8e-62   5.8e-62       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 199.9 bits;  conditional E-value: 5.8e-62\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+L+++ek+++ ++W+kv  +++e Ga++L rl+v+yP+tq+fF++F+dLs + +++g+a+vk+Hgkkv+d +s++l++ld +l++++++LselH++kl+\n",
      "  HBB_SPETO   1 VHLTDGEKNAISTAWGKV--NAAEIGAEALGRLLVVYPWTQRFFDSFGDLSSASAVMGNAKVKAHGKKVIDSFSNGLKHLD-NLKGTFASLSELHCDKLH 97 \n",
      "                69****************..99***********************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++fkll++++v v+a++l+k+ft+++qaa++K++a+va+al++kY+\n",
      "  HBB_SPETO  98 VDPENFKLLGNMIVIVMAHHLGKDFTPEAQAAFQKVVAGVANALSHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBB_SUNMU  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  199.6   0.1   7.4e-62   7.4e-62       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 199.6 bits;  conditional E-value: 7.4e-62\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls +ek  v+ +W+kv  + +evGa++L rl+v+yP+tq+fF++F+dLs + +++g+++vk+Hgkkvl+ l++++a+ld +l++++++LselH++kl+\n",
      "  HBB_SUNMU   1 VHLSGEEKACVTGLWGKV--NEDEVGAEALGRLLVVYPWTQRFFDSFGDLSSASAVMGNPKVKAHGKKVLHSLGEGVANLD-NLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..99***********************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ll++vlvvvla++++keft++vqaa++K++a+va+ala+kY+\n",
      "  HBB_SUNMU  98 VDPENFRLLGNVLVVVLASKFGKEFTPPVQAAFQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBA_MACSI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  198.8   0.5   1.3e-61   1.3e-61       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 198.8 bits;  conditional E-value: 1.3e-61\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+kt+vk +W+kv+ +++e+Ga++L+r+f+s+P+t+++F++F dLs     +gsa+vk Hgkkv+dAl+ a+ ++d +++++l+aLs+lHa+kl+v\n",
      "  HBA_MACSI   1 VLSPADKTNVKDAWGKVGGHAGEYGAEALERMFLSFPTTKTYFPHF-DLS-----HGSAQVKGHGKKVADALTLAVGHVD-DMPQALSALSDLHAHKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+laa+lp+eft++v+a+l+K+la+v+++l+skY+\n",
      "  HBA_MACSI  94 DPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBB_TRIIN  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  198.8   0.1   1.3e-61   1.3e-61       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 198.8 bits;  conditional E-value: 1.3e-61\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+L+++ek  v  +Wakv  +v+e+G+++L rl+v+yP+tq+fFe+F+dLs + +++++++vk+Hg+kv + ++d+l++l+ +l++++++LselH++kl+\n",
      "  HBB_TRIIN   1 VHLTPEEKALVIGLWAKV--NVKEYGGEALGRLLVVYPWTQRFFEHFGDLSSASAIMNNPKVKAHGEKVFTSFGDGLKHLE-DLKGAFAELSELHCDKLH 97 \n",
      "                69****************..***********************************************************98.9***************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ll++vlv+vla +++kef++++qaa++K++a+va+ala+kY+\n",
      "  HBB_TRIIN  98 VDPENFRLLGNVLVCVLARHFGKEFSPEAQAAYQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBB_EQUHE  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  195.9   0.1     1e-60     1e-60       2     149 .]       2     146 .]       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 195.9 bits;  conditional E-value: 1e-60\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                +Ls +ek +v+a+W kv  + eevG+++L rl+v+yP+tq+fF++F+dLs +++++g+++vk+Hgkkvl+ +++++ +ld +l++++++LselH++kl+v\n",
      "  HBB_EQUHE   2 QLSGEEKAAVLALWDKV--NEEEVGGEALGRLLVVYPWTQRFFDSFGDLSNPAAVMGNPKVKAHGKKVLHSFGEGVHHLD-NLKGTFAQLSELHCDKLHV 98 \n",
      "                79***************..99***********************************************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++f+ll++vlvvvla +++k+ft++ qa+++K++a+va+ala+kY+\n",
      "  HBB_EQUHE  99 DPENFRLLGNVLVVVLARHFGKDFTPELQASYQKVVAGVANALAHKYH 146\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA2_GALCR  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  195.2   1.0   1.7e-60   1.7e-60       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 195.2 bits;  conditional E-value: 1.7e-60\n",
      "    globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                 vLs+ +k++vka+W kv+a+++++Ga++L+r+f+s+P+t+++F++F dLs     +gs +vk Hgkkv+dAl++a+ ++d +++++l+aLs+lHa+kl+\n",
      "  HBA2_GALCR   1 VLSPTDKSNVKAAWEKVGAHAGDYGAEALERMFLSFPTTKTYFPHF-DLS-----HGSTQVKGHGKKVADALTNAVLHVD-DMPSALSALSDLHAHKLR 92 \n",
      "                 69********************************************.***.....*************************.****************** PP\n",
      "\n",
      "    globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 vdp++fkll ++l+v+la ++p+eft++v+a+l+K++a+v+++l+skY+\n",
      "  HBA2_GALCR  93 VDPVNFKLLRHCLLVTLACHHPAEFTPAVHASLDKFMASVSTVLTSKYR 141\n",
      "                 ************************************************7 PP\n",
      "\n",
      ">> MYG_LYCPI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  194.7   1.7   2.4e-60   2.4e-60       3     149 .]       2     147 ..       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 194.7 bits;  conditional E-value: 2.4e-60\n",
      "   globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvd 102\n",
      "                Ls++e++ v+++W+kve+d +++G+++L+rlf+ +Pet+++F+kFk+L+tede+kgs+d+kkHg++vl+Al+ +l+k ++++eaelk+L+++Ha+k+k++\n",
      "  MYG_LYCPI   2 LSDGEWQIVLNIWGKVETDLAGHGQEVLIRLFKNHPETLDKFDKFKHLKTEDEMKGSEDLKKHGNTVLTALGGILKK-KGHHEAELKPLAQSHATKHKIP 100\n",
      "                89***************************************************************************.99******************** PP\n",
      "\n",
      "   globins4 103 pkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                +ky++++s+++++vl+++++++f ad +aa++K+l+l+++++a+kYk\n",
      "  MYG_LYCPI 101 VKYLEFISDAIIQVLQNKHSGDFHADTEAAMKKALELFRNDIAAKYK 147\n",
      "                **********************************************7 PP\n",
      "\n",
      ">> MYG_PROGU  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  194.7   0.3   2.4e-60   2.4e-60       3     149 .]       2     147 ..       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 194.7 bits;  conditional E-value: 2.4e-60\n",
      "   globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvd 102\n",
      "                Ls++e++ v++vW+kve d  ++G+++L+rlf+ +Pet+e+F+kFk+L++ede+++s+++kkHg +vl+Al+ +l+k ++++ ael++L+++Ha+k+k++\n",
      "  MYG_PROGU   2 LSDGEWQLVLNVWGKVEGDLSGHGQEVLIRLFKGHPETLEKFDKFKHLKAEDEMRASEELKKHGTTVLTALGGILKK-KGQHAAELAPLAQSHATKHKIP 100\n",
      "                89***************************************************************************.99******************** PP\n",
      "\n",
      "   globins4 103 pkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                +ky++++se++++vl++++p++f+ad+q+a+ K+l+l+++++a+kYk\n",
      "  MYG_PROGU 101 VKYLEFISEAIIQVLQSKHPGDFGADAQGAMSKALELFRNDIAAKYK 147\n",
      "                **********************************************7 PP\n",
      "\n",
      ">> MYG_SAISC  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  193.9   1.1   4.3e-60   4.3e-60       3     149 .]       2     147 ..       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 193.9 bits;  conditional E-value: 4.3e-60\n",
      "   globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvd 102\n",
      "                Ls++e++ v+++W+kvead+  +G+++L+ lf+ +Pet+e+F+kFk+L+ ede+k+s+++kkHg +vl+Al+ +l+k ++++eaelk+L+++Ha+k+k++\n",
      "  MYG_SAISC   2 LSDGEWQLVLNIWGKVEADIPSHGQEVLISLFKGHPETLEKFDKFKHLKSEDEMKASEELKKHGTTVLTALGGILKK-KGQHEAELKPLAQSHATKHKIP 100\n",
      "                89***************************************************************************.99******************** PP\n",
      "\n",
      "   globins4 103 pkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                +ky++l+s+++v+vl++++p++f+ad+q+a++K+l+l+++++a+kYk\n",
      "  MYG_SAISC 101 VKYLELISDAIVHVLQKKHPGDFGADAQGAMKKALELFRNDMAAKYK 147\n",
      "                **********************************************7 PP\n",
      "\n",
      ">> HBB_TACAC  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  192.7   0.4   9.7e-60   9.7e-60       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 192.7 bits;  conditional E-value: 9.7e-60\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls  ekt+v+++W+ v  +v+e G+++L rl+v+yP+tq+fFe+F+dLs +d+++g+a+vk+Hg kvl+ ++dal++ld +l++++++LselH++kl+\n",
      "  HBB_TACAC   1 VHLSGSEKTAVTNLWGHV--NVNELGGEALGRLLVVYPWTQRFFESFGDLSSADAVMGNAKVKAHGAKVLTSFGDALKNLD-NLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ l++vlvvvla +++keft+++qaa +Kl+++v++ala+kY+\n",
      "  HBB_TACAC  98 VDPENFNRLGNVLVVVLARHFSKEFTPEAQAAWQKLVSGVSHALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBE_PONPY  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  190.8   0.6   3.8e-59   3.8e-59       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 190.8 bits;  conditional E-value: 3.8e-59\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+ +++ek +v+++W+k+  +vee G+++L rl+v+yP+tq+fF++F++Ls + ++ g+++vk+Hgkkvl+ ++da++++d +l++++++LselH++kl+\n",
      "  HBE_PONPY   1 VHFTAEEKAAVTSLWSKM--NVEEAGGEALGRLLVVYPWTQRFFDSFGNLSSPSAILGNPKVKAHGKKVLTSFGDAIKNMD-NLKTTFAKLSELHCDKLH 97 \n",
      "                5789**************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++fkll++v+v +la++++keft++vqaa +Kl++ va+ala+kY+\n",
      "  HBE_PONPY  98 VDPENFKLLGNVMVIILATHFGKEFTPEVQAAWQKLVSAVAIALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBA_MESAU  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  190.1   0.4   6.3e-59   6.3e-59       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 190.1 bits;  conditional E-value: 6.3e-59\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+++kt++ ++W+k++ +++e+Ga++L+r+f +yP+t+++F++F d+s     +gsa+vk Hgkkv+dAl++a+ +ld +l+++l+aLs+lHa+kl+v\n",
      "  HBA_MESAU   1 VLSAKDKTNISEAWGKIGGHAGEYGAEALERMFFVYPTTKTYFPHF-DVS-----HGSAQVKGHGKKVADALTNAVGHLD-DLPGALSALSDLHAHKLRV 93 \n",
      "                69********************************************.888.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+la+++p++ft++v+a+l+K++a+v+++l+skY+\n",
      "  HBA_MESAU  94 DPVNFKLLSHCLLVTLANHHPADFTPAVHASLDKFFASVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA2_BOSMU  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  189.5   0.8   9.7e-59   9.7e-59       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 189.5 bits;  conditional E-value: 9.7e-59\n",
      "    globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                 vLs+a+k +vka+W+kv+ +++e+Ga++L+r+f+s+P+t+++F++F dLs     +gsa+vk Hg kv+ Al++a+ +ld +l+++l++Ls+lHa+kl+\n",
      "  HBA2_BOSMU   1 VLSAADKGNVKAAWGKVGGHAAEYGAEALERMFLSFPTTKTYFPHF-DLS-----HGSAQVKGHGAKVAAALTKAVGHLD-DLPGALSELSDLHAHKLR 92 \n",
      "                 69********************************************.***.....*************************.****************** PP\n",
      "\n",
      "    globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 vdp++fklls+ l+v+la++lp++ft++v+a+l+K+la v+++l+skY+\n",
      "  HBA2_BOSMU  93 VDPVNFKLLSHSLLVTLASHLPSDFTPAVHASLDKFLANVSTVLTSKYR 141\n",
      "                 ************************************************7 PP\n",
      "\n",
      ">> HBB_TUPGL  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  188.1   0.0   2.5e-58   2.5e-58       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 188.1 bits;  conditional E-value: 2.5e-58\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls +ek +v+ +W+kv  d e+vG++ L  l+++yP+tq+fF++F+dLs + +++++++vk+Hgkkvl+ +sd+l++ld +l++++++LselH++kl+\n",
      "  HBB_TUPGL   1 VHLSGEEKAAVTGLWGKV--DLEKVGGQSLGSLLIVYPWTQRFFDSFGDLSSPSAVMSNPKVKAHGKKVLTSFSDGLNHLD-NLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ll++vlv vla +++ eft++vqaa++K++a+va+ala+kY+\n",
      "  HBB_TUPGL  98 VDPENFRLLGNVLVRVLACNFGPEFTPQVQAAFQKVVAGVANALAHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBA_AILME  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  187.0   0.2   5.4e-58   5.4e-58       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 187.0 bits;  conditional E-value: 5.4e-58\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+kt+vka W k++ +++e+G+++L+r f s+P+t+++F++F dLs      gsa+vk+Hgkkv+dAl+ a+ +ld +l+++l+aLs+lHa+kl+v\n",
      "  HBA_AILME   1 VLSPADKTNVKATWDKIGGHAGEYGGEALERTFASFPTTKTYFPHF-DLS-----PGSAQVKAHGKKVADALTTAVGHLD-DLPGALSALSDLHAHKLRV 93 \n",
      "                69********************************************.***.....9************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+la+++p+eft++v+a+l+K+++ v+++l+skY+\n",
      "  HBA_AILME  94 DPVNFKLLSHCLLVTLASHHPAEFTPAVHASLDKFFSAVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> MYG_MOUSE  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  186.8   0.3   6.3e-58   6.3e-58       3     149 .]       2     147 ..       1     147 [. 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 186.8 bits;  conditional E-value: 6.3e-58\n",
      "   globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvd 102\n",
      "                Ls++e++ v++vW+kvead +++G+++L+ lf+++Pet+++F+kFk+L+ e+++kgs+d+kkHg +vl+Al+ +l+k ++++ ae+++L+++Ha+k+k++\n",
      "  MYG_MOUSE   2 LSDGEWQLVLNVWGKVEADLAGHGQEVLIGLFKTHPETLDKFDKFKNLKSEEDMKGSEDLKKHGCTVLTALGTILKK-KGQHAAEIQPLAQSHATKHKIP 100\n",
      "                89***************************************************************************.99******************** PP\n",
      "\n",
      "   globins4 103 pkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                +ky++++se++++vl++r++++f+ad+q+a+ K+l+l+++++a+kYk\n",
      "  MYG_MOUSE 101 VKYLEFISEIIIEVLKKRHSGDFGADAQGAMSKALELFRNDIAAKYK 147\n",
      "                **********************************************7 PP\n",
      "\n",
      ">> HBB_ORNAN  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  186.3   0.0   9.1e-58   9.1e-58       1     149 []       1     146 []       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 186.3 bits;  conditional E-value: 9.1e-58\n",
      "   globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                v+Ls +ek++v+++W+kv  +++e G+++L rl+v+yP+tq+fFe F+dLs + +++g+++vk+Hg kvl+ ++dal++ld +l++++++LselH++kl+\n",
      "  HBB_ORNAN   1 VHLSGGEKSAVTNLWGKV--NINELGGEALGRLLVVYPWTQRFFEAFGDLSSAGAVMGNPKVKAHGAKVLTSFGDALKNLD-DLKGTFAKLSELHCDKLH 97 \n",
      "                69****************..*************************************************************.****************** PP\n",
      "\n",
      "   globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                vdp++f+ l++vl+vvla +++k+f+++vqaa +Kl+++va+al +kY+\n",
      "  HBB_ORNAN  98 VDPENFNRLGNVLIVVLARHFSKDFSPEVQAAWQKLVSGVAHALGHKYH 146\n",
      "                ************************************************7 PP\n",
      "\n",
      ">> HBB_COLLI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  185.6   0.1   1.5e-57   1.5e-57       2     149 .]       2     146 .]       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 185.6 bits;  conditional E-value: 1.5e-57\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                + s++ek+ ++++W+kv  +v++ Ga++L rl+++yP+tq+fF +F++Ls + ++ g+++vk+Hgkkvl+ ++da+++ld ++++++++LselH++kl+v\n",
      "  HBB_COLLI   2 HWSAEEKQLITSIWGKV--NVADCGAEALARLLIVYPWTQRFFSSFGNLSSATAISGNPNVKAHGKKVLTSFGDAVKNLD-NIKGTFAQLSELHCDKLHV 98 \n",
      "                789**************..*************************************************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++f+ll+++lv +laa+++k+ft++ qaa +Kl+ +va+ala kY+\n",
      "  HBB_COLLI  99 DPENFRLLGDILVIILAAHFGKDFTPECQAAWQKLVRVVAHALARKYH 146\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBB_LARRI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  181.7   0.1   2.3e-56   2.3e-56       2     149 .]       2     146 .]       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 181.7 bits;  conditional E-value: 2.3e-56\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                + s++ek+ ++ +W+kv  +v++ Ga++L rl+++yP+tq+fF +F++Ls + +++g++ v++Hgkkvl+ +++a+++ld ++++++++LselH++kl+v\n",
      "  HBB_LARRI   2 HWSAEEKQLITGLWGKV--NVADCGAEALARLLIVYPWTQRFFASFGNLSSPTAINGNPMVRAHGKKVLTSFGEAVKNLD-NIKNTFAQLSELHCDKLHV 98 \n",
      "                789**************..*************************************************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++f+ll+++l+ vlaa+++k+ft+d qaa +Kl+ +va+ala kY+\n",
      "  HBB_LARRI  99 DPENFRLLGDILIIVLAAHFAKDFTPDSQAAWQKLVRVVAHALARKYH 146\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA_PAGLA  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  180.8   0.7   4.4e-56   4.4e-56       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 180.8 bits;  conditional E-value: 4.4e-56\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs a+k+++ka W k++++++e+Ga++L+r f+s+P+t+++F++F dLs     +gsa+vk+Hgkkv+dAl+ a+ +l+ +l+++l+aLs+lHa kl+v\n",
      "  HBA_PAGLA   1 VLSSADKNNIKATWDKIGSHAGEYGAEALERTFISFPTTKTYFPHF-DLS-----HGSAQVKAHGKKVADALTLAVGHLE-DLPNALSALSDLHAYKLRV 93 \n",
      "                69********************************************.***.....***********************98.9****************** PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+la ++p+eft++v++al+K+++ v+++l+skY+\n",
      "  HBA_PAGLA  94 DPVNFKLLSHCLLVTLACHHPAEFTPAVHSALDKFFSAVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA_ERIEU  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  179.3   0.2   1.4e-55   1.4e-55       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 179.3 bits;  conditional E-value: 1.4e-55\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+ +k +vk+ W+k + + +e+G+++L r+f+++P+t+++F++F dL+      gsa+vk Hgkkv+dAl+ a+++ld +++++l+aLs+lHa+kl+v\n",
      "  HBA_ERIEU   1 VLSATDKANVKTFWGKLGGHGGEYGGEALDRMFQAHPTTKTYFPHF-DLN-----PGSAQVKGHGKKVADALTTAVNNLD-DVPGALSALSDLHAHKLRV 93 \n",
      "                69********************************************.999.....9************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+la ++p++ft++v+a+l+K+la+va++l+skY+\n",
      "  HBA_ERIEU  94 DPVNFKLLSHCLLVTLALHHPADFTPAVHASLDKFLATVATVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA_PROLO  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  179.2   0.1   1.4e-55   1.4e-55       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 179.2 bits;  conditional E-value: 1.4e-55\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+k ++ka W k++ +++e+G+++L+r f s+P+t+++F++F dLs      gsa+vk+Hgkkv+dAl+ a+ +ld +l+++l+aLs+lHa kl+v\n",
      "  HBA_PROLO   1 VLSPADKANIKATWDKIGGHAGEYGGEALERTFASFPTTKTYFPHF-DLS-----PGSAQVKAHGKKVADALTLAVGHLD-DLPGALSALSDLHAYKLRV 93 \n",
      "                69********************************************.***.....9************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fklls++l+v+la ++p+eft++v+a+l+K++ +v+++l+skY+\n",
      "  HBA_PROLO  94 DPVNFKLLSHCLLVTLACHHPAEFTPAVHASLDKFFTSVSTVLTSKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBB1_VAREX  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  168.7   0.1   2.5e-52   2.5e-52       2     149 .]       2     146 .]       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 168.7 bits;  conditional E-value: 2.5e-52\n",
      "    globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                 + +++ek+ + ++W+k+  dv+  G++ L  l+v yP+tq+ F +F++Ls + ++ g++ vk+Hgkkvl+ ++da+++ld +++ ++++LselH++kl+\n",
      "  HBB1_VAREX   2 HWTAEEKQLICSLWGKI--DVGLIGGETLAGLLVIYPWTQRQFSHFGNLSSPTAIAGNPRVKAHGKKVLTSFGDAIKNLD-NIKDTFAKLSELHCDKLH 97 \n",
      "                 6799*************..*************************************************************.****************** PP\n",
      "\n",
      "    globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 vdp +fkll++vlv vla +++keft++ +aa++Kl+++v++ la +Y+\n",
      "  HBB1_VAREX  98 VDPTNFKLLGNVLVIVLADHHGKEFTPAHHAAYQKLVNVVSHSLARRYH 146\n",
      "                 ************************************************7 PP\n",
      "\n",
      ">> HBAZ_HORSE  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  167.9   0.2   4.2e-52   4.2e-52       3     149 .]       2     141 .]       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 167.9 bits;  conditional E-value: 4.2e-52\n",
      "    globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                 L++ae+t v ++W+k+  +++ vG+++L+rlf syP+t+++F++F dL+     +gs+++++Hg kv+ A++da++++d ++ ++l++LselHa  l+v\n",
      "  HBAZ_HORSE   2 LTKAERTMVVSIWGKISMQADAVGTEALQRLFSSYPQTKTYFPHF-DLH-----EGSPQLRAHGSKVAAAVGDAVKSID-NVAGALAKLSELHAYILRV 93 \n",
      "                 89*******************************************.999.....9************************.******************* PP\n",
      "\n",
      "    globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 dp++fk+ls++l+v+la+rlp++ftad++aa +K+l++v+ +l++kY+\n",
      "  HBAZ_HORSE  94 DPVNFKFLSHCLLVTLASRLPADFTADAHAAWDKFLSIVSSVLTEKYR 141\n",
      "                 ***********************************************7 PP\n",
      "\n",
      ">> HBA_FRAPO  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  165.5   0.5   2.4e-51   2.4e-51       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 165.5 bits;  conditional E-value: 2.4e-51\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+k++vk +++k+ +++e++Ga++L+r+f++yP t+++F++F dLs     +gsa+vk Hgkkv+ Al +a +++d ++ ++l++Ls+lHa+kl+v\n",
      "  HBA_FRAPO   1 VLSAADKNNVKGIFGKISSHAEDYGAEALERMFITYPSTKTYFPHF-DLS-----HGSAQVKGHGKKVVAALIEAANHID-DIAGTLSKLSDLHAHKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fkll+++ +vv+a ++p+ +t++v+a+l+K+l  v ++l++kY+\n",
      "  HBA_FRAPO  94 DPVNFKLLGQCFLVVVAIHHPSALTPEVHASLDKFLCAVGNVLTAKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA_PHACO  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  164.9   0.5   3.6e-51   3.6e-51       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 164.9 bits;  conditional E-value: 3.6e-51\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+k++vk +++k+  ++ee+Ga++L+r+f++yP t+++F++F dLs     +gsa++k Hgkkv+ Al +a++++d ++ ++l++Ls+lHa+kl+v\n",
      "  HBA_PHACO   1 VLSAADKNNVKGIFTKIAGHAEEYGAEALERMFITYPSTKTYFPHF-DLS-----HGSAQIKGHGKKVVAALIEAVNHID-DITGTLSKLSDLHAHKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fkll+++ +vv+a ++p+ +t++v+a+l+K+l  v ++l++kY+\n",
      "  HBA_PHACO  94 DPVNFKLLGQCFLVVVAIHHPSALTPEVHASLDKFLCAVGTVLTAKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBA_ANSSE  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  160.7   0.6     7e-50     7e-50       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 160.7 bits;  conditional E-value: 7e-50\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+a+k +vk+v++k++ ++ee+Ga+ L+r+f+++P+t+++F++F dL+      gsa++k+Hgkkv+ Al +a +++d ++ ++l++Ls+lHa+kl+v\n",
      "  HBA_ANSSE   1 VLSAADKGNVKTVFGKIGGHAEEYGAETLQRMFQTFPQTKTYFPHF-DLQ-----PGSAQIKAHGKKVAAALVEAANHID-DIAGALSKLSDLHAQKLRV 93 \n",
      "                69********************************************.999.....9************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fk+l+++ +vvla ++p+ +t++v+a+++K+l  va++l++kY+\n",
      "  HBA_ANSSE  94 DPVNFKFLGHCFLVVLAIHHPSLLTPEVHASMDKFLCAVATVLTAKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBAD_PASMO  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  160.6   0.1   7.6e-50   7.6e-50       3     149 .]       2     141 .]       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 160.6 bits;  conditional E-value: 7.6e-50\n",
      "    globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                 L++++k+ ++++W+k +   ee Gad+L r+f syP t+++F++F dLs     +gs++++ Hgkkv+ Als+a+++ld +l+++l++Ls+lHa +l+v\n",
      "  HBAD_PASMO   2 LTAEDKKLIQQIWGKLGGAEEEIGADALWRMFHSYPSTKTYFPHF-DLS-----QGSDQIRGHGKKVVAALSNAIKNLD-NLSQALSELSNLHAYNLRV 93 \n",
      "                 99*******************************************.***.....9************************.******************* PP\n",
      "\n",
      "    globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 dp++fk+ls++l v la+rl+ke++++v++a++K+++ va +la+kY+\n",
      "  HBAD_PASMO  94 DPVNFKFLSQCLQVSLATRLGKEYSPEVHSAVDKFMSAVASVLAEKYR 141\n",
      "                 ***********************************************7 PP\n",
      "\n",
      ">> HBA_TRIOC  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  160.1   0.4   1.1e-49   1.1e-49       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 160.1 bits;  conditional E-value: 1.1e-49\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+++kt+vk+v++k+  ++e++Ga+ L+r+f++yP t+++F++F dL+     +gsa++k+Hgkkv++Al +a++++d ++ ++l++Ls+lHa+kl+v\n",
      "  HBA_TRIOC   1 VLSANDKTNVKTVFTKITGHAEDYGAETLERMFITYPPTKTYFPHF-DLH-----HGSAQIKAHGKKVVGALIEAVNHID-DIAGALSKLSDLHAQKLRV 93 \n",
      "                69********************************************.999.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fkll+++ +vv+a ++p+ +t++v+a+l+K+l  v ++l++kY+\n",
      "  HBA_TRIOC  94 DPVNFKLLGQCFLVVVAIHHPSVLTPEVHASLDKFLCAVGNVLSAKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBAD_CHLME  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  159.6   0.5   1.5e-49   1.5e-49       3     149 .]       2     141 .]       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 159.6 bits;  conditional E-value: 1.5e-49\n",
      "    globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                 L++++k+ ++++W kv  + ee G+++L+r+f++yP+t+++F++F dL+      gs++v+ Hgkkv+ Al++a+++ld +l+++l++Ls+lHa +l+v\n",
      "  HBAD_CHLME   2 LTADDKKLLTQLWEKVAGHQEEFGSEALQRMFLTYPQTKTYFPHF-DLH-----PGSEQVRGHGKKVAAALGNAVKSLD-NLSQALSELSNLHAYNLRV 93 \n",
      "                 89*******************************************.999.....9************************.******************* PP\n",
      "\n",
      "    globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 dp +fkll+++  vvla++l+k+++++++aa++K+l+ va +la+kY+\n",
      "  HBAD_CHLME  94 DPANFKLLAQCFQVVLATHLGKDYSPEMHAAFDKFLSAVAAVLAEKYR 141\n",
      "                 ***********************************************7 PP\n",
      "\n",
      ">> HBA_COLLI  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  159.6   0.3   1.6e-49   1.6e-49       2     149 .]       1     141 []       1     141 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 159.6 bits;  conditional E-value: 1.6e-49\n",
      "   globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                vLs+++k++vkav+ak++ ++++ G+++L+rlf++yP+t+++F++F dLs     +gsa++k Hgkkv++Al +a +++d ++ ++l++Ls+lHa+kl+v\n",
      "  HBA_COLLI   1 VLSANDKSNVKAVFAKIGGQAGDLGGEALERLFITYPQTKTYFPHF-DLS-----HGSAQIKGHGKKVAEALVEAANHID-DIAGALSKLSDLHAQKLRV 93 \n",
      "                69********************************************.***.....*************************.******************* PP\n",
      "\n",
      "   globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                dp++fkll+++ +vv+a ++p+ +t++v+a+l+K++  v ++l++kY+\n",
      "  HBA_COLLI  94 DPVNFKLLGHCFLVVVAVHFPSLLTPEVHASLDKFVLAVGTVLTAKYR 141\n",
      "                ***********************************************7 PP\n",
      "\n",
      ">> HBBL_RANCA  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  158.6   0.3   3.1e-49   3.1e-49       2     149 .]       2     146 .]       1     146 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 158.6 bits;  conditional E-value: 3.1e-49\n",
      "    globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                 + +++ek  +++vW kv  dve+ G ++L+rlf++yP+tq++F  F+dLs ++++ g+++v++Hgkk+l+A+ +a+ +ld +++++l++Lse Ha++l+\n",
      "  HBBL_RANCA   2 HWTAEEKAVINSVWQKV--DVEQDGHEALTRLFIVYPWTQRYFSTFGDLSSPAAIAGNPKVHAHGKKILGAIDNAIHNLD-DVKGTLHDLSEEHANELH 97 \n",
      "                 6799*************..*************************************************************.****************** PP\n",
      "\n",
      "    globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 vdp++f+ l+evl+vvl a+l+k f+++vq   eK++a++  al++ Y+\n",
      "  HBBL_RANCA  98 VDPENFRRLGEVLIVVLGAKLGKAFSPQVQHVWEKFIAVLVDALSHSYH 146\n",
      "                 ************************************************7 PP\n",
      "\n",
      ">> HBB2_XENTR  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  155.8   1.2   2.3e-48   2.3e-48       2     148 ..       2     145 ..       1     146 [] 0.98\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 155.8 bits;  conditional E-value: 2.3e-48\n",
      "    globins4   2 vLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklk 100\n",
      "                 + +++ek ++ +vW+kv  d+e+ G d+L rl+v+yP+tq++F +F++Ls   ++ g+ +vk+Hg+kvl A++ a+++ld +++++lk Ls++Ha+ l+\n",
      "  HBB2_XENTR   2 HWTAEEKATIASVWGKV--DIEQDGHDALSRLLVVYPWTQRYFSSFGNLSNVSAVSGNVKVKAHGNKVLSAVGSAIQHLD-DVKSHLKGLSKSHAEDLH 97 \n",
      "                 6799*************..*************************************************************.****************** PP\n",
      "\n",
      "    globins4 101 vdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskY 148\n",
      "                 vdp++fk l++vlv vlaa+l++ ft++vqa  eKl a++  al++ Y\n",
      "  HBB2_XENTR  98 VDPENFKRLADVLVIVLAAKLGSAFTPQVQAVWEKLNATLVAALSHGY 145\n",
      "                 ********************************************9988 PP\n",
      "\n",
      ">> HBA4_SALIR  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  141.1   0.1   7.6e-44   7.6e-44       3     149 .]       2     142 .]       1     142 [] 0.98\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 141.1 bits;  conditional E-value: 7.6e-44\n",
      "    globins4   3 LseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkv 101\n",
      "                 Ls+++k +vka+W+k+  + +e G+++L r++v+yP+t+ +F ++  +       gsa vkkHg ++++ + d + ++d +l + l +LselHa+kl+v\n",
      "  HBA4_SALIR   2 LSAKDKANVKAIWGKILPKSDEIGEQALSRMLVVYPQTKAYFSHWASVA-----PGSAPVKKHGITIMNQIDDCVGHMD-DLFGFLTKLSELHATKLRV 94 \n",
      "                 9********************************************7777.....9************************.9****************** PP\n",
      "\n",
      "    globins4 102 dpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                 dp +fk+l++ l+vv+aa +p+eft++++ +++K+l+ +a ala+kY+\n",
      "  HBA4_SALIR  95 DPTNFKILAHNLIVVIAAYFPAEFTPEIHLSVDKFLQQLALALAEKYR 142\n",
      "                 ***********************************************7 PP\n",
      "\n",
      ">> MYG_MUSAN  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  120.1   0.4   2.3e-37   2.3e-37       7     149 .]       2     142 ..       1     142 [. 0.98\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 120.1 bits;  conditional E-value: 2.3e-37\n",
      "   globins4   7 ektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakklkvdpkyf 106\n",
      "                +++kv++vW+ ve+d    G++iL rlf  yPe+q++F+kFk+ s   elk  ad+k+ + +vl Al+++++k ++ +++ +kaL+  H + +k++p+yf\n",
      "  MYG_MUSAN   2 DWEKVNSVWSAVESDLTAIGQNILLRLFEQYPESQNHFPKFKNKS-LGELKDTADIKAQADTVLSALGNIVKK-KGSHSQPVKALAATHITTHKIPPHYF 99 \n",
      "                89***************************************9877.99*************************.99************************ PP\n",
      "\n",
      "   globins4 107 kllsevlvvvlaarlpkeftadvqaaleKllalvakalaskYk 149\n",
      "                  ++ + v+vl + +p+e+ a+vqaa+  ++++++ ++ + Yk\n",
      "  MYG_MUSAN 100 TKITTIAVDVLSEMYPSEMNAQVQAAFSGAFKIICSDIEKEYK 142\n",
      "                ****************************************997 PP\n",
      "\n",
      ">> HBB2_TRICR  \n",
      "   #    score  bias  c-Evalue  i-Evalue hmmfrom  hmm to    alifrom  ali to    envfrom  env to     acc\n",
      " ---   ------ ----- --------- --------- ------- -------    ------- -------    ------- -------    ----\n",
      "   1 !  117.9   0.0   1.1e-36   1.1e-36       1     148 [.       1     145 []       1     145 [] 0.99\n",
      "\n",
      "  Alignments for each domain:\n",
      "  == domain 1  score: 117.9 bits;  conditional E-value: 1.1e-36\n",
      "    globins4   1 vvLseaektkvkavWakveadveevGadiLvrlfvsyPetqefFekFkdLstedelkgsadvkkHgkkvldAlsdalakldekleaelkaLselHakkl 99 \n",
      "                 v+L++++++++ a+ +kv  +v+  G++ L rl+v+ P+++++F  F+dLs  d++  +++v +Hg kv+  + +a ++ld +l++ +++Ls  H+ k+\n",
      "  HBB2_TRICR   1 VHLTAEDRKEIAAILGKV--NVDSLGGQCLARLIVVNPWSRRYFHDFGDLSSCDAICRNPKVLAHGAKVMRSIVEATKHLD-NLREYYADLSVTHSLKF 96 \n",
      "                 69****************..*************************************************************.***************** PP\n",
      "\n",
      "    globins4 100 kvdpkyfkllsevlvvvlaarlpkeftadvqaaleKllalvakalaskY 148\n",
      "                  vdp++fkl+s +++v la  l ++f+ + q a+eKl+++v++al + Y\n",
      "  HBB2_TRICR  97 YVDPENFKLFSGIVIVCLALTLQTDFSCHKQLAFEKLMKGVSHALGHGY 145\n",
      "                 *********************************************9988 PP\n",
      "\n",
      "\n",
      "\n",
      "Internal pipeline statistics summary:\n",
      "-------------------------------------\n",
      "Query model(s):                            1  (149 nodes)\n",
      "Target sequences:                         45  (6519 residues searched)\n",
      "Passed MSV filter:                        45  (1); expected 0.9 (0.02)\n",
      "Passed bias filter:                       45  (1); expected 0.9 (0.02)\n",
      "Passed Vit filter:                        45  (1); expected 0.0 (0.001)\n",
      "Passed Fwd filter:                        45  (1); expected 0.0 (1e-05)\n",
      "Initial search space (Z):                 45  [actual number of targets]\n",
      "Domain search space  (domZ):              45  [number of targets reported over threshold]\n",
      "# CPU time: 0.01u 0.01s 00:00:00.02 Elapsed: 00:00:00.02\n",
      "# Mc/sec: 40.87\n",
      "//\n",
      "[ok]\n"
     ]
    }
   ],
   "source": [
    "!/Users/dml/.local/share/mamba/envs/myjlabenv/bin/hmmsearch globins4.hmm globins45.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "d25b587a-8c2c-49fb-a4c5-bd6bb81ee2cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/dml/JupyterLab/hmmer-3.4/tutorial\n"
     ]
    }
   ],
   "source": [
    "print(os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "d45a1ef5-921e-4ebb-ad08-4539c4d6048f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "curl: Remote file name has no length\n",
      "curl: (23) Failed writing received data to disk/application\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: curl failed. File may have moved again.\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\n# 1. Download the current human protein FASTA file using the new, robust NCBI path.\\n# -O: names the local output file.\\ncurl -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\\n\\n# Check if the download was successful before proceeding\\nif [ $? -ne 0 ]; then\\n    echo \"Error: curl failed. File may have moved again.\"\\n    exit 1\\nfi\\n\\n# 2. Uncompress the downloaded file\\ngunzip human.protein.faa.gz\\n\\n# 3. Count the lines in the uncompressed file (to confirm the download worked)\\nwc -l human.protein.faa\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mCalledProcessError\u001b[39m                        Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[50]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mbash\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 1. Download the current human protein FASTA file using the new, robust NCBI path.\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# -O: names the local output file.\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mcurl -O human.protein.faa.gz \u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# Check if the download was successful before proceeding\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mif [ $? -ne 0 ]; then\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m    echo \u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mError: curl failed. File may have moved again.\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m    exit 1\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mfi\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 2. Uncompress the downloaded file\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mgunzip human.protein.faa.gz\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 3. Count the lines in the uncompressed file (to confirm the download worked)\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mwc -l human.protein.faa\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/interactiveshell.py:2565\u001b[39m, in \u001b[36mInteractiveShell.run_cell_magic\u001b[39m\u001b[34m(self, magic_name, line, cell)\u001b[39m\n\u001b[32m   2563\u001b[39m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m.builtin_trap:\n\u001b[32m   2564\u001b[39m     args = (magic_arg_s, cell)\n\u001b[32m-> \u001b[39m\u001b[32m2565\u001b[39m     result = \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   2567\u001b[39m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[32m   2568\u001b[39m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[32m   2569\u001b[39m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[32m   2570\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic.MAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/magics/script.py:160\u001b[39m, in \u001b[36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[39m\u001b[34m(line, cell)\u001b[39m\n\u001b[32m    158\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m    159\u001b[39m     line = script\n\u001b[32m--> \u001b[39m\u001b[32m160\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/magics/script.py:348\u001b[39m, in \u001b[36mScriptMagics.shebang\u001b[39m\u001b[34m(self, line, cell)\u001b[39m\n\u001b[32m    343\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m args.raise_error \u001b[38;5;129;01mand\u001b[39;00m p.returncode != \u001b[32m0\u001b[39m:\n\u001b[32m    344\u001b[39m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[32m    345\u001b[39m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[32m    346\u001b[39m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[32m    347\u001b[39m     rc = p.returncode \u001b[38;5;129;01mor\u001b[39;00m -\u001b[32m9\u001b[39m\n\u001b[32m--> \u001b[39m\u001b[32m348\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[31mCalledProcessError\u001b[39m: Command 'b'\\n# 1. Download the current human protein FASTA file using the new, robust NCBI path.\\n# -O: names the local output file.\\ncurl -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\\n\\n# Check if the download was successful before proceeding\\nif [ $? -ne 0 ]; then\\n    echo \"Error: curl failed. File may have moved again.\"\\n    exit 1\\nfi\\n\\n# 2. Uncompress the downloaded file\\ngunzip human.protein.faa.gz\\n\\n# 3. Count the lines in the uncompressed file (to confirm the download worked)\\nwc -l human.protein.faa\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# 1. Download the current human protein FASTA file using the new, robust NCBI path.\n",
    "# -O: names the local output file.\n",
    "curl -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\n",
    "\n",
    "# Check if the download was successful before proceeding\n",
    "if [ $? -ne 0 ]; then\n",
    "    echo \"Error: curl failed. File may have moved again.\"\n",
    "    exit 1\n",
    "fi\n",
    "\n",
    "# 2. Uncompress the downloaded file\n",
    "gunzip human.protein.faa.gz\n",
    "\n",
    "# 3. Count the lines in the uncompressed file (to confirm the download worked)\n",
    "wc -l human.protein.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62a378aa-bc6a-498a-9494-0ee02cb4b668",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "4fed2aec-576b-4175-8c33-73528b71d70e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 4: wget: command not found\n",
      "gunzip: can't stat: human.protein.faa.gz (human.protein.faa.gz.gz): No such file or directory\n",
      "wc: human.protein.faa: open: No such file or directory\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\n# 1. Download the current human protein FASTA file using the correct RefSeq path.\\n# This path points to the GRCh38 reference assembly in the RefSeq database.\\nwget -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\\n\\n# 2. Uncompress the downloaded file\\ngunzip human.protein.faa.gz\\n\\n# 3. Count the lines in the uncompressed file (to confirm the download worked)\\nwc -l human.protein.faa\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mCalledProcessError\u001b[39m                        Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[51]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[43mget_ipython\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mrun_cell_magic\u001b[49m\u001b[43m(\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mbash\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 1. Download the current human protein FASTA file using the correct RefSeq path.\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# This path points to the GRCh38 reference assembly in the RefSeq database.\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mwget -O human.protein.faa.gz \u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 2. Uncompress the downloaded file\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mgunzip human.protein.faa.gz\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m# 3. Count the lines in the uncompressed file (to confirm the download worked)\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43mwc -l human.protein.faa\u001b[39;49m\u001b[38;5;130;43;01m\\n\u001b[39;49;00m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/interactiveshell.py:2565\u001b[39m, in \u001b[36mInteractiveShell.run_cell_magic\u001b[39m\u001b[34m(self, magic_name, line, cell)\u001b[39m\n\u001b[32m   2563\u001b[39m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m.builtin_trap:\n\u001b[32m   2564\u001b[39m     args = (magic_arg_s, cell)\n\u001b[32m-> \u001b[39m\u001b[32m2565\u001b[39m     result = \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   2567\u001b[39m \u001b[38;5;66;03m# The code below prevents the output from being displayed\u001b[39;00m\n\u001b[32m   2568\u001b[39m \u001b[38;5;66;03m# when using magics with decorator @output_can_be_silenced\u001b[39;00m\n\u001b[32m   2569\u001b[39m \u001b[38;5;66;03m# when the last Python token in the expression is a ';'.\u001b[39;00m\n\u001b[32m   2570\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(fn, magic.MAGIC_OUTPUT_CAN_BE_SILENCED, \u001b[38;5;28;01mFalse\u001b[39;00m):\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/magics/script.py:160\u001b[39m, in \u001b[36mScriptMagics._make_script_magic.<locals>.named_script_magic\u001b[39m\u001b[34m(line, cell)\u001b[39m\n\u001b[32m    158\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m    159\u001b[39m     line = script\n\u001b[32m--> \u001b[39m\u001b[32m160\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mshebang\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/.local/share/mamba/envs/myjlabenv/lib/python3.14/site-packages/IPython/core/magics/script.py:348\u001b[39m, in \u001b[36mScriptMagics.shebang\u001b[39m\u001b[34m(self, line, cell)\u001b[39m\n\u001b[32m    343\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m args.raise_error \u001b[38;5;129;01mand\u001b[39;00m p.returncode != \u001b[32m0\u001b[39m:\n\u001b[32m    344\u001b[39m     \u001b[38;5;66;03m# If we get here and p.returncode is still None, we must have\u001b[39;00m\n\u001b[32m    345\u001b[39m     \u001b[38;5;66;03m# killed it but not yet seen its return code. We don't wait for it,\u001b[39;00m\n\u001b[32m    346\u001b[39m     \u001b[38;5;66;03m# in case it's stuck in uninterruptible sleep. -9 = SIGKILL\u001b[39;00m\n\u001b[32m    347\u001b[39m     rc = p.returncode \u001b[38;5;129;01mor\u001b[39;00m -\u001b[32m9\u001b[39m\n\u001b[32m--> \u001b[39m\u001b[32m348\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(rc, cell)\n",
      "\u001b[31mCalledProcessError\u001b[39m: Command 'b'\\n# 1. Download the current human protein FASTA file using the correct RefSeq path.\\n# This path points to the GRCh38 reference assembly in the RefSeq database.\\nwget -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\\n\\n# 2. Uncompress the downloaded file\\ngunzip human.protein.faa.gz\\n\\n# 3. Count the lines in the uncompressed file (to confirm the download worked)\\nwc -l human.protein.faa\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# 1. Download the current human protein FASTA file using the correct RefSeq path.\n",
    "# This path points to the GRCh38 reference assembly in the RefSeq database.\n",
    "wget -O human.protein.faa.gz \"ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/all_assembly_versions/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_protein.faa.gz\"\n",
    "\n",
    "# 2. Uncompress the downloaded file\n",
    "gunzip human.protein.faa.gz\n",
    "\n",
    "# 3. Count the lines in the uncompressed file (to confirm the download worked)\n",
    "wc -l human.protein.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "101aed05-5214-4261-8c96-f46cf5a58ed6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-10-19 13:04:22--  ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/mRNA_Prot/human.protein.faa.gz\n",
      "           => ‘human.protein.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.13, 130.14.250.31, 130.14.250.7, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.13|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /refseq/H_sapiens/mRNA_Prot ... done.\n",
      "==> SIZE human.protein.faa.gz ... done.\n",
      "\n",
      "==> PASV ... done.    ==> RETR human.protein.faa.gz ... \n",
      "No such file ‘human.protein.faa.gz’.\n",
      "\n",
      "gunzip: can't stat: human.protein.faa.gz (human.protein.faa.gz.gz): No such file or directory\n",
      "wc: human.protein.faa: open: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!/Users/dml/.local/share/mamba/envs/myjlabenv/bin/wget ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/mRNA_Prot/human.protein.faa.gz ; \\\n",
    " /usr/bin/gunzip human.protein.faa.gz ; \\\n",
    " /usr/bin/wc -l human.protein.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "46843d17-c452-493a-9399-0085a2cc8fdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-10-19 13:04:26--  ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/mRNA_Prot/human.protein.faa.gz\n",
      "           => ‘human.protein.faa.gz’\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.11, 130.14.250.12, 130.14.250.13, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /refseq/H_sapiens/mRNA_Prot ... done.\n",
      "==> SIZE human.protein.faa.gz ... done.\n",
      "\n",
      "==> PASV ... done.    ==> RETR human.protein.faa.gz ... \n",
      "No such file ‘human.protein.faa.gz’.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Users/dml/.local/share/mamba/envs/myjlabenv/bin/wget ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/mRNA_Prot/human.protein.faa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bdf48482-c41e-4670-8fa2-66e1f7abce76",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biopython kernel",
   "language": "python",
   "name": "myjlabenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
